### PR TITLE
[KVStore] Integration test for version upgrade

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR adds an integration test for the protocol state version upgrade process.
- Removes duplicate `InvalidEpochTransitionAttempted` field from KVStore

### Remaining TODOs
- When Snapshot contains protocol state ID:
  - [x] Pass in v0 version of model in bootstrapping ([code](https://github.com/onflow/flow-go/pull/5477/files#diff-bf93c45a15b46de12072db8fb5962e3a8d4d6abc6adde04ea09d88ce6c639b61R258-R262))  
  - [x] Validate protocol state version in snapshot rather than against database ([code](https://github.com/onflow/flow-go/pull/5477/files#diff-d9f96296953d5decbde39a805649e291768d8bddb40bd14ea83a05820d5b8fe1R25-R32))
  - [x] Remove `InvalidEpochTransitionAttempted` field from kv store (duplicated from epoch protocol state entry)
  - [x] Reduce wait time for version changes in tests
  - [ ] Don’t inspect service events directly for metrics emission ([code](https://github.com/onflow/flow-go/pull/5477/files#diff-c79d3fc5d32f153917b7cd923ab4ba64ef22f12268ddd13f353a06391f68a96dR880)) -> added TODO to https://github.com/onflow/flow-go/issues/5666
- When branch is updated to use Cadence 1.0
  - [ ] Rewrite transaction in C1.0 syntax ([code](https://github.com/onflow/flow-go/pull/5477/files#diff-8d3755aaefae34250b3b99e000adc3e806817b21abc9665f7f009af0999c75ac))

### CODE REVIEW

File: state/protocol/badger/mutator.go [hunk 1]
```diff
@@ -877,6 +877,7 @@ func (m *FollowerState) epochTransitionMetricsAndEventsOnBlockFinalized(block *f
 // Suppose an EpochSetup service event is emitted during execution of block A. C seals A, therefore
 // we apply the metrics/events when C is finalized. The first block of the EpochSetup
 // phase is block C.
+// TODO: this should read data from the KVStore/ProtocolStateEntry for the block rather than parsing service events
```
Comment (AlexHentschel) [inline state/protocol/badger/mutator.go:880]: lets capture this in an issue. I am worried that if we forget that, there might be an unnecessary pager duty alert going out

**Review by AlexHentschel — COMMENTED** [review]
very impressive test -- I liked how compact/highly-modular the testing code is for verifying 2 upgraders from initial to supported to unsupported version 👏

**Review by AlexHentschel — APPROVED** [review]

File: state/protocol/badger/mutator.go (hunk 1 above)
Comment (jordanschalm) [reply state/protocol/badger/mutator.go:880]: 👍  It's captured in https://github.com/onflow/flow-go/issues/5666.

File: cmd/bootstrap/cmd/finalize.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/onflow/flow-go/state/protocol/protocol_state/kvstore"
```
Comment (durkmurder) [inline cmd/bootstrap/cmd/finalize.go:11]: nit: should group it with other flow-go entries.

File: integration/tests/lib/block_state.go [hunk 3]
```diff
@@ -263,6 +310,28 @@ func (bs *BlockState) WaitForSealed(t *testing.T, height uint64) *flow.Block {
...
+			}
+			return bs.highestSealed != nil && bs.highestSealed.Header.View >= view
+		},
+		blockStateTimeout,
+		100*time.Millisecond,
+		fmt.Sprintf("did not receive sealed block for view (%v) within %v seconds", view, blockStateTimeout))
+
+	bs.RLock()
+	defer bs.RUnlock()
+	return bs.highestSealed
```
Comment (durkmurder) [inline integration/tests/lib/block_state.go:332]: How about using `bs.HighestSealed()` here and in `WaitForSealedHeight`

File: state/protocol/inmem/convert.go [hunk 4]
```diff
@@ -4,6 +4,8 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/onflow/flow-go/state/protocol/protocol_state"
+
 	"github.com/onflow/flow-go/model/encodable"
```
Comment (durkmurder) [inline state/protocol/inmem/convert.go:9]:
```suggestion
	"github.com/onflow/flow-go/state/protocol/protocol_state"
	"github.com/onflow/flow-go/model/encodable"
```

**Review by durkmurder — APPROVED** [review]
Very nice, I like how compact and awaiting style your test is.

**Pull request merged**